### PR TITLE
Add anonymous bounty toggle

### PR DESCRIPTION
**Pull request by yasmasdas**
Add anonymous bounty toggle

No idea who "as4567" is (I guess misconfigured github desktop)

### CODE REVIEW

**Review by Yimura — DISMISSED** [review]

File: src/views/players/player/player_troll.cpp [hunk 1]
```diff
@@ -17,12 +17,15 @@ namespace big
 			components::player_command_button<"rcplayer">(g_player_service->get_selected());
 
 			static int bounty_value = 0;
-
+			static bool anonymous = true;
```
Comment (tupoy-ya) [inline src/views/players/player/player_troll.cpp]: Would be nice to save this.

**Review by tupoy-ya — DISMISSED** [review]

File: src/views/players/player/player_troll.cpp (hunk 1 above)
Comment (yasmasdas) [reply src/views/players/player/player_troll.cpp]: And how would I do that? @tupoy-ya

Comment (tupoy-ya) [reply src/views/players/player/player_troll.cpp]: You can look at how `breakupcheating` is implemented, it does the same for the breakup message.
https://github.com/YimMenu/YimMenu/blob/081dd6d69418c7b79c1c6ad812bf389c0d872e1f/src/backend/commands/player/kick/breakup_kick.cpp#L65

**Comment by yasmasdas** [comment]
@tupoy-ya Is that what you mean? I'm little confused

File: src/backend/commands/player/troll/set_bounty.cpp [hunk 2]
```diff
@@ -11,9 +11,9 @@ namespace big
 
 		virtual void execute(player_ptr player, const std::vector<std::uint64_t>& _args, const std::shared_ptr<command_context> ctx)
 		{
-			troll::set_bounty_on_player(player, 10000);
+			troll::set_bounty_on_player(player, 10000, true);
```
Comment (tupoy-ya) [inline src/backend/commands/player/troll/set_bounty.cpp]: Why this doesn't use `g.session.anonymous_bounty`?

```suggestion
			troll::set_bounty_on_player(player, 10000, g.session.anonymous_bounty);
```

File: src/backend/commands/player/troll/set_bounty.cpp [hunk 3]
```diff
@@ -11,9 +11,9 @@ namespace big
...
 		}
 	};
 
 	set_bounty g_bounty("bounty", "Bounty", "Sets a 10k bounty on the player", 0);
```
Comment (tupoy-ya) [inline src/backend/commands/player/troll/set_bounty.cpp:19]:
```suggestion
	set_bounty g_bounty("bounty", "Bounty", "Sets a 10k bounty on the player", 0);
	bool_command g_anonymous_bounty("anonbounty", "Anonymous Bounty", "Don't show your name when setting a bounty on someone", g.session.anonymous_bounty);
```

File: src/views/players/player/player_troll.cpp [hunk 4]
```diff
@@ -17,12 +17,14 @@ namespace big
 			components::player_command_button<"rcplayer">(g_player_service->get_selected());
 
 			static int bounty_value = 0;
-
+			
 			ImGui::SliderInt("Bounty", &bounty_value, 0, 10000);
 			ImGui::SameLine();
-			components::button("Set", [] { troll::set_bounty_on_player(g_player_service->get_selected(), bounty_value);});
+			ImGui::Checkbox("Anonymous Bounty", &g.session.anonymous_bounty);
```
Comment (tupoy-ya) [inline src/views/players/player/player_troll.cpp]:
```suggestion
			components::command_checkbox<"anonbounty">();
```

**Review by tupoy-ya — COMMENTED** [review]
Well mostly, but why not use a command for this toggle instead?

Suggested changes are untested.

**Comment by yasmasdas** [comment]
thanks for that
should be good for review now?

**Review by Yimura — DISMISSED** [review]

**Comment by Yimura** [comment]
Going to merge this even though it conflicts with a recent PR, they'll have to adjust their code accordingly.

**Comment by Yimura** [comment]
@yasmasdas could you fix the compiler warning? CI won't build correctly currently:
image

**Comment by Yimura** [comment]
Doubt that'll fix it, it's best to just work locally on your machine than coding on Github web.
We've got time, I'm not in a hurry to merge this.

**Comment by yasmasdas** [comment]
looks like it did fix it

**Review by Yimura — APPROVED** [review]

**Pull request merged**